### DISCUSSION TITLE
tabs

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#822] [Ideas]
Re-arranging https://github.com/ddkwork/crypt the interface of the repository with core found that when specifying the orientation of the tab widget to be vertical, the orientation of the tab widget should not be changed after the window is widened

**Comment by ddkwork** [comment]
```go
func main2() {
	b := gi.NewAppBody("crypt tools")
	tabs := gi.NewTabs(b)
	tabs.Style(func(s *styles.Style) {
		s.Direction = styles.Row
	})
	v1 := tabs.NewTab("v1")
	x1 := gi.NewTabs(v1)
	x1.NewTab("des")
	x1.NewTab("rsa")
	x1.NewTab("rsa")
	x1.NewTab("rsa")
	x1.NewTab("rsa")
	x1.NewTab("rsa")
	x1.NewTab("rsa")
	x1.NewTab("rsa")
	x1.NewTab("rsa")
	x1.NewTab("rsa")

	tabs.NewTab("v2")
	tabs.NewTab("v3")
	tabs.NewTab("v4")
	tabs.NewTab("v5")
	tabs.NewTab("v6")

	b.NewWindow().Run().Wait()
}

```

**Reply by kkoreilly** [reply]
To customize the appearance/orientation of the tabs, you should first set the tab type to one of the standardized ones if it meets your needs (`gi.NewTabs(b).SetType(...)`). Beyond that, you can use custom styling. Also, there is currently a bug with tab type `NavigationAuto` that I am going to fix today.

**Reply by ddkwork** [reply]
> To customize the appearance/orientation of the tabs, you should first set the tab type to one of the standardized ones if it meets your needs (`gi.NewTabs(b).SetType(...)`). Beyond that, you can use custom styling. Also, there is currently a bug with tab type `NavigationAuto` that I am going to fix today.

i will try, thx you.

**Reply by ddkwork** [reply]
> To customize the appearance/orientation of the tabs, you should first set the tab type to one of the standardized ones if it meets your needs (`gi.NewTabs(b).SetType(...)`). Beyond that, you can use custom styling. Also, there is currently a bug with tab type `NavigationAuto` that I am going to fix today.

Working now, thx.

**Comment by ddkwork** [comment]
![Uploading screenshots.gif…]()
